gensim/models/poincare.py

```diff
@@ -167,47 +171,48 @@ def __init__(self, train_data, size=50, alpha=0.1, negative=10, workers=1, epsil
         self._np_random = np_random.RandomState(seed)
         self.init_range = init_range
         self._loss_grad = None
-        self._load_relations()
-        self._init_embeddings()
+        self.build_vocab(train_data)
 
-    def _load_relations(self):
+    def build_vocab(self, relations=None, update=False):
         """Load relations from the train data and build vocab."""
-        vocab = {}
-        index2word = []
-        all_relations = []  # List of all relation pairs
-        node_relations = defaultdict(set)  # Mapping from node index to its related node indices
+        old_index2word_len = len(self.kv.index2word)
 
         logger.info("loading relations from train data..")
-        for relation in self.train_data:
+        for relation in relations:
```

Comment:
I think we risk tripping over a TypeError here. You're iterating over something that can be None.

Please double-check and add a unit test to cover this case, if necessary.

Reply:
Thank you for the nice review.
Because we must input some relations (not None) in build_vocab, I removed default None value here 👍